libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1728,10 +1728,39 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 	else
 		_operation.subExpression().accept(*this);
 	Type const* subExprType = type(_operation.subExpression());
-	TypeResult result = type(_operation.subExpression())->unaryOperatorResult(op);
-	if (!result)
+
+
+	// Check if the operator is built-in or user-defined.
+	Result<FunctionDefinition const*> userDefinedOperatorResult = subExprType->userDefinedOperator(
+		_operation.getOperator(),
+		*currentDefinitionScope(),
+		true
+	);
+	if (userDefinedOperatorResult)
+		_operation.annotation().userDefinedFunction = userDefinedOperatorResult;
```

Comment:
Well, maybe. I really did find it unnecessarily hard to read at first because of the naming. It's probably because the name tells me mostly where the value came from (it's a result of `userDefinedOperatorResult()`, rather than what it is (definition of the operator), which makes me jump back and forth to get that info.

Can you rename it to `operatorDefinitionResult` now that the function has been renamed? I think it would help a bit.